src/include/mutex.h

```diff
@@ -1,11 +1,13 @@
 #ifndef CMARK_MUTEX_H
 #define CMARK_MUTEX_H
 
+#include "cmark-gfm_config.h"
+
 #include <stdbool.h>
 
-#ifdef CMARK_THREADING
+#if CMARK_THREADING
 
-#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
+#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__)) || defined(__wasi__)
 #include <unistd.h>
 #endif
```

Comment:
At this point i'm starting to wonder if we can just get away with just checking for the existence of `unistd.h` instead of probing for different compilation environments:

```suggestion
#if defined(__has_include)
#if __has_include(<unistd.h>)
#include <unistd.h>
#endif
#endif
```

It looks like both Clang and GCC have the `__has_include` macro, so this should probably cover our bases for any other new platform over time.

Reply:
@QuietMisdreavus I really appreciate your review.

As you said, using the `__has_include` macro makes the code more robust against future changes, but I hesitate to use it.

Because the `__has_include` macro is standard in C++17 and C23, but not in C99. This repository appears to be compiled as C99, at least in CMake.

https://github.com/swiftlang/swift-cmark/blob/b97d09472e847a416629f026eceae0e2afcfad65/CMakeLists.txt#L16-L18

Reply:
Oh heck, i didn't realize it was tied to a C standard. It's likely the C99 requirement is inherited from GitHub-Flavored Markdown (or even CommonMark) so i would hesitate to change it.

In that case, we can merge this as-is and live with the ever-growing list of Unix-like platforms here. 😅